core-api/src/main/java/com/optimizely/ab/internal/ExperimentUtils.java

```diff
@@ -65,23 +67,32 @@ public static boolean isUserInExperiment(@Nonnull ProjectConfig projectConfig,
             Boolean resolveReturn = resolveAudienceConditions(projectConfig, experiment, attributes);
             return resolveReturn == null ? false : resolveReturn;
         }
+        else {
+            Boolean resolveReturn = resolveAudience(projectConfig, experiment, attributes);
+            return resolveReturn == null ? false : resolveReturn;
+        }
+    }
+
+    public static @Nullable Boolean resolveAudience(@Nonnull ProjectConfig projectConfig,
+                                                    @Nonnull Experiment experiment,
+                                                    @Nonnull Map<String, ?> attributes) {
         List<String> experimentAudienceIds = experiment.getAudienceIds();
 
         // if there are no audiences, ALL users should be part of the experiment
         if (experimentAudienceIds.isEmpty()) {
             return true;
         }
 
+        List<Condition> conditions = new ArrayList<>();
         for (String audienceId : experimentAudienceIds) {
-            AudienceIdCondition conditions = new AudienceIdCondition(projectConfig.getAudience(audienceId));
-            Boolean conditionEval = conditions.evaluate(attributes);
-
-            if (conditionEval != null && conditionEval) {
-                return true;
-            }
+            AudienceIdCondition condition = new AudienceIdCondition(audienceId);
+            conditions.add(condition);
         }
 
-        return false;
+        OrCondition implicitOr = new OrCondition(conditions);
+
+        return implicitOr.evaluate(projectConfig, attributes);
+
```

Comment:
Short and sweet! I guess I'm fine with restricting `experiment.audienceIds` to being an array of strings, but FYI I'm still going to [beat the drum](https://optimizely.atlassian.net/browse/OASIS-3642) to ensure that we infer `"or"` instead of `"not"` in contexts where we're parsing non-leaf nodes in a full condition tree (e.g. `experiment.audienceConditions` and `audience.conditions`). Doing so will help ensure that all our SDKs/snippets behave identically and can't accidentally negate any conditions.

Reply:
Oh, I might be missing something, but I'm talking about places like [this](https://github.com/optimizely/java-sdk/blob/da3b24f9c8078722a4bc03cd7fd99b16a1b12552/core-api/src/main/java/com/optimizely/ab/config/parser/JsonSimpleConfigParser.java#L322-L323)

Reply:
yes, they are all fixed in this PR.  default is Or

Reply:
Ah, sorry, I think I'd minimized those diffs! My bad. Thanks so much for doing this - looks good overall, though I have a couple comments which I'll make inline 👍